ViewControllers/NewsFeedViewController.m

```diff
@@ -57,41 +58,48 @@ -(void)fetchNews{
                 NSString *newsDescription = news.title;
                 NSLog(@": YD: %@", newsDescription);
             }
-            NSUserDefaults *userDefaults = [NSUserDefaults standardUserDefaults];
-            NSMutableArray *stocksForPersonalization = [[userDefaults arrayForKey:@"Soi"] mutableCopy];
-            Stock *stock = stocksForPersonalization[0];
-
-            NSLog(@": StocksOfInterest: %@", stock.ticker);
         } else {
             NSLog(@"Error getting News Feed: %@", error.localizedDescription);
         }
         [self.newsFeedTableView reloadData];
+        //[self updateToPersonalizedNews];
         [self.refresh endRefreshing];
     }];
 }
 
 - (void) updateToPersonalizedNews{
-    if (self.stocksOfInterest.count != 0){
-        Stock *keyStock = [self.stocksOfInterest objectAtIndex:self.stocksOfInterest.count - 1];
-        NSString *keyword = keyStock.ticker;
-        [[APIManager shared] fetchHeadlineNews:(NSString *) keyword completion:^(NSArray *allNewsArticles, NSError *error) {
-            if (allNewsArticles) {
-                for (NSMutableDictionary *dict in allNewsArticles) {
-                    [self.newsArray insertObject:dict atIndex:0];
-                }
-                NSLog(@"Successfully loaded Headline News");
-                //
-                for (News *news in allNewsArticles) {
-                    // uses text field in stock model to fetch the text body of a stock.
-                    NSString *newsDescription = news.title;
-                    NSLog(@": YD: %@", newsDescription);
+    NSMutableArray *keywords = [[NSMutableArray alloc] init];
+    [[PFUser query] getFirstObjectInBackgroundWithBlock:^(PFObject *object, NSError *error)
+    {
+        PFUser *currentUser = [PFUser currentUser];
+        [keywords addObjectsFromArray:[currentUser valueForKey:@"StocksOfInterest"]];
+        NSString *key = [keywords lastObject];
+        NSLog(@"TICKERS: %@", key);
+        if (keywords != nil){
+            [[APIManager shared] fetchHeadlineNews:(NSString *) key completion:^(NSMutableArray *keywordArticles, NSError *error) {
+                if (keywordArticles) {
+                    //[self.newsArray addObjectsFromArray:keywordArticles]
+                    NSLog(@"Successfully loaded Headline News");
+                    for (News *news in keywordArticles) {
+                        // uses text field in stock model to fetch the text body of a stock.
+                        NSString *newsDescription = news.title;
+                        NSLog(@": YD UPDATED: %@", newsDescription);
+                    }
+                    for (News *dict in keywordArticles) {
+                        [self.newsArray insertObject:dict atIndex:0];
+                        NSLog(@": Updated News Descriptions: %@", dict.description);
+                    }
```

Comment:
It should be possible to simplify this code by using a `NSMutableArray` function (after removing the `NSLog()`